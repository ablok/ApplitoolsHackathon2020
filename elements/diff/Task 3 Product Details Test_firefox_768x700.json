[
  {
    ".access_link": [
      {
        "op": "update",
        "path": [0, "location", "x"],
        "val": 695,
        "oldVal": 635
      }
    ]
  },
  {
    ".wishlist": [
      {
        "op": "update",
        "path": [0, "location", "x"],
        "val": 711,
        "oldVal": 681
      }
    ]
  },
  {
    "#shoe_img": [
      {
        "op": "update",
        "path": [0, "css", "backgroundImage"],
        "val": "none",
        "oldVal": "url(\"https://demo.applitools.com/grid/img/products/shoes/1.jpg\")"
      }
    ]
  },
  {
    ".prod_info em": [
      {
        "op": "update",
        "path": [0, "location", "x"],
        "val": 75,
        "oldVal": 105
      }
    ]
  },
  {
    ".prod_info p small": [
      {
        "op": "update",
        "path": [0, "css", "textDecoration"],
        "val": "rgb(255, 255, 255)",
        "oldVal": "rgb(68, 68, 68)"
      },
      {
        "op": "update",
        "path": [0, "css", "color"],
        "val": "rgb(255, 255, 255)",
        "oldVal": "rgb(68, 68, 68)"
      }
    ]
  },
  {
    ".custom-select-form li": [
      {
        "op": "update",
        "path": [0, "text"],
        "val": "Not Small",
        "oldVal": "Small (S)"
      }
    ]
  },
  {
    ".btn_add_to_cart a": [
      {
        "op": "update",
        "path": [0, "location", "y"],
        "val": 836.0999755859375,
        "oldVal": 886.0999755859375
      }
    ]
  },
  {
    "#new_price": [
      {
        "op": "update",
        "path": [0, "location", "x"],
        "val": 187.8999938964844,
        "oldVal": 165.23333740234375
      },
      {
        "op": "update",
        "path": [0, "size", "width"],
        "val": 27.300003051757812,
        "oldVal": 49.96665954589844
      },
      {
        "op": "update",
        "path": [0, "text"],
        "val": "$33",
        "oldVal": "$33.00"
      }
    ]
  }
]
