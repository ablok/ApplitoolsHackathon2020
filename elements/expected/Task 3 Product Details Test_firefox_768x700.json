[
  {
    "#logo img": [
      {
        "visibility": true,
        "location": {
          "x": 287.54998779296875,
          "y": 16.300003051757812
        },
        "size": {
          "width": 192.88333129882812,
          "height": 20
        },
        "css": {
          "display": "inline",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(0, 77, 218)",
          "color": "rgb(0, 77, 218)",
          "font": ""
        },
        "image": "/grid/img/logo.svg"
      }
    ]
  },
  {
    ".main-menu": [
      {
        "visibility": false,
        "location": {
          "x": -1536,
          "y": 0
        },
        "size": {
          "width": 345.6000061035156,
          "height": 626
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgb(255, 255, 255)",
          "backgroundImage": "none",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        }
      }
    ]
  },
  {
    "//div[@class='hamburger-box']/ancestor::a": [
      {
        "visibility": true,
        "location": {
          "x": 15,
          "y": 10.300003051757812
        },
        "size": {
          "width": 26,
          "height": 32
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(0, 77, 218)",
          "color": "rgb(0, 77, 218)",
          "font": ""
        }
      }
    ]
  },
  {
    ".custom-search-input": [
      {
        "visibility": true,
        "location": {
          "x": 202,
          "y": 51.30000305175781
        },
        "size": {
          "width": 364,
          "height": 40
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgb(242, 242, 242)",
          "backgroundImage": "none",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        }
      }
    ]
  },
  {
    ".btn_search_mob": [
      {
        "visibility": false,
        "location": {
          "x": 0,
          "y": 0
        },
        "size": {
          "width": 0,
          "height": 0
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(17, 17, 17)",
          "color": "rgb(17, 17, 17)",
          "font": ""
        }
      }
    ]
  },
  {
    ".access_link": [
      {
        "visibility": true,
        "location": {
          "x": 635,
          "y": 51.30000305175781
        },
        "size": {
          "width": 26,
          "height": 60
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(17, 17, 17)",
          "color": "rgb(17, 17, 17)",
          "font": ""
        }
      }
    ]
  },
  {
    ".wishlist": [
      {
        "visibility": true,
        "location": {
          "x": 681,
          "y": 51.30000305175781
        },
        "size": {
          "width": 26,
          "height": 60
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(17, 17, 17)",
          "color": "rgb(17, 17, 17)",
          "font": ""
        }
      }
    ]
  },
  {
    ".cart_bt": [
      {
        "visibility": true,
        "location": {
          "x": 727,
          "y": 51.30000305175781
        },
        "size": {
          "width": 26,
          "height": 60
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(17, 17, 17)",
          "color": "rgb(17, 17, 17)",
          "font": ""
        }
      }
    ]
  },
  {
    ".cart_bt strong": [
      {
        "visibility": true,
        "location": {
          "x": 740,
          "y": 85.30000305175781
        },
        "size": {
          "width": 16,
          "height": 16
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgb(8, 52, 135)",
          "backgroundImage": "none",
          "textDecoration": "rgb(255, 255, 255)",
          "color": "rgb(255, 255, 255)",
          "font": ""
        },
        "text": "2"
      }
    ]
  },
  {
    "#shoe_img": [
      {
        "visibility": true,
        "location": {
          "x": 15,
          "y": 201.1000061035156
        },
        "size": {
          "width": 738,
          "height": 350
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "url(\"https://demo.applitools.com/grid/img/products/shoes/1.jpg\")",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        }
      }
    ]
  },
  {
    ".prod_info .icon-star": [
      {
        "visibility": true,
        "location": {
          "x": 15,
          "y": 643.0999755859375
        },
        "size": {
          "width": 18,
          "height": 17
        },
        "css": {
          "display": "inline",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(254, 195, 72)",
          "color": "rgb(254, 195, 72)",
          "font": ""
        }
      },
      {
        "visibility": true,
        "location": {
          "x": 33,
          "y": 643.0999755859375
        },
        "size": {
          "width": 18,
          "height": 17
        },
        "css": {
          "display": "inline",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(254, 195, 72)",
          "color": "rgb(254, 195, 72)",
          "font": ""
        }
      },
      {
        "visibility": true,
        "location": {
          "x": 51,
          "y": 643.0999755859375
        },
        "size": {
          "width": 18,
          "height": 17
        },
        "css": {
          "display": "inline",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(254, 195, 72)",
          "color": "rgb(254, 195, 72)",
          "font": ""
        }
      },
      {
        "visibility": true,
        "location": {
          "x": 69,
          "y": 643.0999755859375
        },
        "size": {
          "width": 18,
          "height": 17
        },
        "css": {
          "display": "inline",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(254, 195, 72)",
          "color": "rgb(254, 195, 72)",
          "font": ""
        }
      },
      {
        "visibility": true,
        "location": {
          "x": 87,
          "y": 643.0999755859375
        },
        "size": {
          "width": 18,
          "height": 17
        },
        "css": {
          "display": "inline",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(136, 135, 135)",
          "color": "rgb(136, 135, 135)",
          "font": ""
        }
      }
    ]
  },
  {
    ".prod_info em": [
      {
        "visibility": true,
        "location": {
          "x": 105,
          "y": 643.0999755859375
        },
        "size": {
          "width": 58.600006103515625,
          "height": 17
        },
        "css": {
          "display": "inline",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(136, 135, 135)",
          "color": "rgb(136, 135, 135)",
          "font": ""
        },
        "text": "4 reviews"
      }
    ]
  },
  {
    ".prod_info p small": [
      {
        "visibility": true,
        "location": {
          "x": 15,
          "y": 681.0999755859375
        },
        "size": {
          "width": 84.75,
          "height": 14
        },
        "css": {
          "display": "inline",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        },
        "text": "SKU: MTKRY-001"
      }
    ]
  },
  {
    ".prod_info p": [
      {
        "visibility": true,
        "location": {
          "x": 15,
          "y": 677.0999755859375
        },
        "size": {
          "width": 738,
          "height": 84
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        }
      }
    ]
  },
  {
    "#quantity_1": [
      {
        "visibility": true,
        "location": {
          "x": 556,
          "y": 820.0999755859375
        },
        "size": {
          "width": 40,
          "height": 40
        },
        "css": {
          "display": "inline-block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(0, 0, 0)",
          "color": "rgb(0, 0, 0)",
          "font": ""
        },
        "value": "1"
      }
    ]
  },
  {
    ".custom-select-form .list": [
      {
        "visibility": true,
        "location": {
          "x": 444,
          "y": 807.8499755859375
        },
        "size": {
          "width": 264,
          "height": 96
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgb(255, 255, 255)",
          "backgroundImage": "none",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        }
      }
    ]
  },
  {
    ".custom-select-form li": [
      {
        "visibility": true,
        "location": {
          "x": 444,
          "y": 807.8499755859375
        },
        "size": {
          "width": 264,
          "height": 24
        },
        "css": {
          "display": "list-item",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        },
        "text": "Small (S)"
      },
      {
        "visibility": true,
        "location": {
          "x": 444,
          "y": 831.8499755859375
        },
        "size": {
          "width": 264,
          "height": 24
        },
        "css": {
          "display": "list-item",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        },
        "text": "M"
      },
      {
        "visibility": true,
        "location": {
          "x": 444,
          "y": 855.8499755859375
        },
        "size": {
          "width": 264,
          "height": 24
        },
        "css": {
          "display": "list-item",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        },
        "text": "L"
      },
      {
        "visibility": true,
        "location": {
          "x": 444,
          "y": 879.8499755859375
        },
        "size": {
          "width": 264,
          "height": 24
        },
        "css": {
          "display": "list-item",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        },
        "text": "XL"
      }
    ]
  },
  {
    ".btn_add_to_cart a": [
      {
        "visibility": true,
        "location": {
          "x": 631.88330078125,
          "y": 885.0999755859375
        },
        "size": {
          "width": 121.11666870117188,
          "height": 41
        },
        "css": {
          "display": "inline-block",
          "backgroundColor": "rgb(0, 77, 218)",
          "backgroundImage": "none",
          "textDecoration": "rgb(255, 255, 255)",
          "color": "rgb(255, 255, 255)",
          "font": ""
        },
        "text": "Add to Cart"
      }
    ]
  },
  {
    "#new_price": [
      {
        "visibility": true,
        "location": {
          "x": 165.23333740234375,
          "y": 897.5999755859375
        },
        "size": {
          "width": 49.96665954589844,
          "height": 19
        },
        "css": {
          "display": "inline",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(0, 77, 218)",
          "color": "rgb(0, 77, 218)",
          "font": ""
        },
        "text": "$33.00"
      }
    ]
  },
  {
    "#old_price": [
      {
        "visibility": true,
        "location": {
          "x": 218.6666717529297,
          "y": 895.0999755859375
        },
        "size": {
          "width": 49.96665954589844,
          "height": 24
        },
        "css": {
          "display": "inline-block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "line-through rgb(153, 153, 153)",
          "color": "rgb(153, 153, 153)",
          "font": ""
        },
        "text": "$48.00"
      }
    ]
  },
  {
    "#discount": [
      {
        "visibility": true,
        "location": {
          "x": 282.1000061035156,
          "y": 896.0999755859375
        },
        "size": {
          "width": 86.89999389648438,
          "height": 20
        },
        "css": {
          "display": "inline-block",
          "backgroundColor": "rgb(255, 51, 51)",
          "backgroundImage": "none",
          "textDecoration": "rgb(255, 255, 255)",
          "color": "rgb(255, 255, 255)",
          "font": ""
        },
        "text": "-30% discount"
      }
    ]
  },
  {
    "//h3[text()='Quick Links']": [
      {
        "visibility": true,
        "location": {
          "x": 15,
          "y": 1021.0999755859376
        },
        "size": {
          "width": 354,
          "height": 16.79998779296875
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(255, 255, 255)",
          "color": "rgb(255, 255, 255)",
          "font": ""
        },
        "text": "Quick Links"
      }
    ]
  },
  {
    "#collapse_1": [
      {
        "visibility": true,
        "location": {
          "x": 15,
          "y": 1052.9000244140625
        },
        "size": {
          "width": 354,
          "height": 152.8000030517578
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        }
      }
    ]
  },
  {
    "//h3[text()='Contacts']": [
      {
        "visibility": true,
        "location": {
          "x": 399,
          "y": 1021.0999755859376
        },
        "size": {
          "width": 354,
          "height": 16.79998779296875
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(255, 255, 255)",
          "color": "rgb(255, 255, 255)",
          "font": ""
        },
        "text": "Contacts"
      }
    ]
  },
  {
    "#collapse_3": [
      {
        "visibility": true,
        "location": {
          "x": 399,
          "y": 1052.9000244140625
        },
        "size": {
          "width": 354,
          "height": 78
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        }
      }
    ]
  },
  {
    "//h3[text()='Keep in touch']": [
      {
        "visibility": true,
        "location": {
          "x": 15,
          "y": 1230.699951171875
        },
        "size": {
          "width": 354,
          "height": 16.800003051757812
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(255, 255, 255)",
          "color": "rgb(255, 255, 255)",
          "font": ""
        },
        "text": "Keep in touch"
      }
    ]
  },
  {
    "#collapse_4": [
      {
        "visibility": true,
        "location": {
          "x": 15,
          "y": 1262.5
        },
        "size": {
          "width": 354,
          "height": 40
        },
        "css": {
          "display": "block",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "backgroundImage": "none",
          "textDecoration": "rgb(68, 68, 68)",
          "color": "rgb(68, 68, 68)",
          "font": ""
        }
      }
    ]
  }
]
